### PR TITLE
Image button

### PR DESCRIPTION
**Pull request by Darius3001**

### CODE REVIEW

File: code/button/ImageButton.xml [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
+    android:layout_width="80dp"
+    android:layout_height="80dp"
+    android:background="?selectableItemBackgroundBorderless"
     android:contentDescription="@string/example_image_button"
     app:layout_constraintBottom_toBottomOf="parent"
     app:layout_constraintEnd_toEndOf="parent"
     app:layout_constraintStart_toStartOf="parent"
     app:layout_constraintTop_toTopOf="parent"
-    app:srcCompat="@android:drawable/btn_star"
+    app:srcCompat="@drawable/c24logo"
```
Comment (Stallinator) [inline code/button/ImageButton.xml:11]: include the glide stuff here :)

File: demo/src/main/java/de/check24/demo/features/button/ComposableImageButtonActivity.kt [hunk 2]
```diff
@@ -52,14 +57,18 @@ class ComposableImageButtonActivity : ComponentActivity() {
     }
 }
 
+@OptIn(ExperimentalMaterialApi::class)
 @Composable
 private fun ImageButton() {
-    Button(
+    IconButton(
         onClick = {},
+        modifier = Modifier.size(240.dp)
```
Comment (Stallinator) [inline demo/src/main/java/de/check24/demo/features/button/ComposableImageButtonActivity.kt:65]: lets look for better fix for ripple effect here

**Review by Stallinator — CHANGES_REQUESTED** [review]
see comments

File: demo/src/main/java/de/check24/demo/features/button/ComposableImageButtonActivity.kt [hunk 3]
```diff
@@ -52,14 +57,18 @@ class ComposableImageButtonActivity : ComponentActivity() {
...
+
     ) {
-        Icon(
-            imageVector = Icons.Rounded.Star,
-            contentDescription = "Image of Button"
+        Image(
```
Comment (Tautellini) [inline demo/src/main/java/de/check24/demo/features/button/ComposableImageButtonActivity.kt:68]: While this works, an IconButton is supposed to have an Icon as Composable Content.

File: demo/src/main/java/de/check24/demo/features/button/AndroidUIImageButtonActivity.kt [hunk 4]
```diff
@@ -2,12 +2,22 @@ package de.check24.demo.features.button
...
 import de.check24.demo.R
 
 class AndroidUIImageButtonActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
         setContentView(R.layout.image_button)
         supportActionBar?.title = "ImageButton"
+
+        val imageView = findViewById<android.widget.ImageButton>(R.id.imageButton)
```
Comment (Tautellini) [inline demo/src/main/java/de/check24/demo/features/button/AndroidUIImageButtonActivity.kt:15]: Im not sure what we decided for the demo project, but this should be done the compose/hybrid way. We have a glide example in our internal code base.

File: code/button/ImageButton.xml (hunk 1 above)
Comment (Stallinator) [reply code/button/ImageButton.xml:11]: remove the srcCompat line pls

File: demo/src/main/res/layout/image_button.xml [hunk 5]
```diff
@@ -6,13 +6,14 @@
...
+        android:layout_width="80dp"
+        android:layout_height="80dp"
+        android:background="?selectableItemBackgroundBorderless"
         android:contentDescription="@string/example_image_button"
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toTopOf="parent"
-        app:srcCompat="@android:drawable/btn_star" />
+        app:srcCompat="@drawable/c24logo" />
```
Comment (Stallinator) [inline demo/src/main/res/layout/image_button.xml:17]: remove the src compat line

**Review by Stallinator — APPROVED** [review]

**Pull request merged**